task-13/src/main/java/com/epam/winter_java_lab/task_13/controller/MessageController.java

```diff
@@ -0,0 +1,105 @@
+package com.epam.winter_java_lab.task_13.controller;
+
+import com.epam.winter_java_lab.task_13.domain.Message;
+import com.epam.winter_java_lab.task_13.domain.User;
+import com.epam.winter_java_lab.task_13.dto.MessageDto;
+import com.epam.winter_java_lab.task_13.service.Impl.MessageService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Set;
+
+@RestController
+@RequestMapping("/api")
+@Api(value = "message resources") //swagger
+public class MessageController {
+
+    private MessageService messageService;
+
+    @Autowired
+    public MessageController(MessageService messageService) {
+        this.messageService = messageService;
+    }
+
+    @GetMapping("/")
+    @ApiOperation(value = "show latest posts", response = Iterable.class)
+    public List<MessageDto> greeting() {
+        return messageService.getMessagesByCreatedDateTime();
+    }
+
+    @GetMapping("/main")
+    @ApiOperation(value = "show messages by filter", response = Iterable.class)
+    public List<MessageDto> main(
+            @RequestParam(required = false, defaultValue = "") String filter,
+            @PageableDefault(sort = {"id"}, direction = Sort.Direction.DESC) Pageable pageable) {
```

Comment:
You should use another approach to customize pageable in swagger 3. I provided appropriate example